contracts/strategies/traderJoe/strategy-joe-avax-ethe.sol

```diff
@@ -1,99 +1,123 @@
-        // I understand the possibility of being frontrun
-        // But AVAX is a dark forest, and I wanna see how this plays out
-        // i.e. will be be heavily frontrunned?
-        //      if so, a new strategy will be deployed.
-
-        // Collects Joe tokens
-        IMasterChefJoeV2(masterChefJoeV2).deposit(poolId, 0);
-
-        uint256 _joe = IERC20(joe).balanceOf(address(this));
-        if (_joe > 0) {
-            // 10% is sent to treasury
-            uint256 _keep = _joe.mul(keep).div(keepMax);
-            uint256 _amount = _joe.sub(_keep).div(2);
-            if (_keep > 0) {
-                _takeFeeJoeToSnob(_keep);
-            }
-            IERC20(joe).safeApprove(joeRouter, 0);
-            IERC20(joe).safeApprove(joeRouter, _joe.sub(_keep));
 
-            _swapTraderJoe(joe, wavax, _amount);
-            _swapTraderJoe(joe, eth, _amount);
+        // ** Harvest our rewards from masterchef ** // 
+        /// @param _pid is the pool id for the lp tokens
+        /// @param _amount is the amount to be deposited into masterchef
+        IMasterChefJoe(masterchefJoe).deposit(lpPoolId, 0);
+
+        // ** Wraps any AVAX that might be present into wavax ** //
+        uint256 _avax = address(this).balance;                 
+        if (_avax > 0) {                                       
+            WAVAX(wavax).deposit{value: _avax}();
         }
 
-        // Adds in liquidity for AVAX/ETH
-        uint256 _wavax = IERC20(wavax).balanceOf(address(this));
+        // ** Swap all our reward tokens for wavax ** //
+        uint256 _joe = IERC20(joe).balanceOf(address(this));            // get balance of joe tokens
+        if(_joe > 0) {
+            _swapToWavax(joe, _joe);
+        }
 
-        uint256 _eth = IERC20(eth).balanceOf(address(this));
+        // ** Takes the fee and swaps for equal shares in our lp token ** // 
+        uint256 _wavax = IERC20(wavax).balanceOf(address(this));            
+        if (_wavax > 0) {
+            uint256 _keep = _wavax.mul(keep).div(keepMax); 
+            if (_keep > 0) {
+               _takeFeeWavaxToSnob(_keep); 
+
+               _wavax = IERC20(wavax).balanceOf(address(this));
+               _swapTraderJoe(wavax, wethe, _wavax.div(2));                     
+            }
+        }
 
-        if (_wavax > 0 && _eth > 0) {
+        // ** Adds liqudity for the AVAX-WETHE LP ** ///
+        _wavax = IERC20(wavax).balanceOf(address(this));
+        uint256 _wethe = IERC20(wethe).balanceOf(address(this));
+        if (_wavax > 0 && _wethe > 0) {
             IERC20(wavax).safeApprove(joeRouter, 0);
             IERC20(wavax).safeApprove(joeRouter, _wavax);
 
-            IERC20(eth).safeApprove(joeRouter, 0);
-            IERC20(eth).safeApprove(joeRouter, _eth);
-
+            IERC20(wethe).safeApprove(joeRouter, 0);
+            IERC20(wethe).safeApprove(joeRouter, _wethe);
+            
+            /// @param tokenA is the first token in the lp
```

Comment:
Maybe we could move all the `@param` headers to the `IJoeRouter` contract and use `@dev` to reference that?